testing/cli/test_cli_sct_get_centerline.py

```diff
@@ -31,5 +33,54 @@ def test_sct_get_centerline_output_file_exists_with_o_arg(tmp_path, ext):
+    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '1', '-o',
+                                  os.path.join(tmp_path, 't2_seg_centerline_soft'+ext), '-qc', str(tmp_path)])
+    for file in [os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz'), os.path.join(tmp_path, 't2_seg_centerline_soft.csv')]:
+        assert os.path.exists(file)
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_get_centerline_soft_output_is_equal_to_one(tmp_path):
+    """This test checks that the sum of the output intensities of the soft centerline is equal to 1 on all slices. using default usage of the CLI
+    script."""
+    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '1', '-o',
+                                  os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz')])
+
+    # Read nii file
+    im = Image(os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz'))
+    # Sum soft centerline intensities in the (x,y) plane, across all slices
+    sum_over_slices = np.apply_over_axes(np.sum, im.data, [0, 2]).flatten()
+    # Test if the summation for each slice is equal to 1
+    assert (sum_over_slices == 1).all()
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_get_centerline_soft_overlaps_with_binary(tmp_path):
+    """This test checks that the output maximum of the soft centerline overlaps with the binary segmentation on all slices."""
+    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '0', '-o',
+                                  os.path.join(tmp_path, 't2_seg_centerline_bin.nii.gz')])
+    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '1', '-o',
+                                  os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz')])
+
+    # Read nii file
+    im_soft = Image(os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz'))  # Read soft centerline file
+    im_bin = Image(os.path.join(tmp_path, 't2_seg_centerline_bin.nii.gz'))  # Read binary centerline file
+    # Find the maximum intensity voxel across all slices in soft centerline and binary centerline
+    max_over_slices_soft = np.apply_over_axes(np.max, im_soft.data, [0, 2])
+    max_over_slices_bin = np.apply_over_axes(np.max, im_bin.data, [0, 2])
+    # Find the coordinates of the maximum in each slice in both soft and binary centerline
+    max_coords_over_slices_soft = np.transpose(np.where(im_soft.data == max_over_slices_soft))
+    max_coords_over_slices_bin = np.transpose(np.where(im_bin.data == max_over_slices_bin))
+    # Test if the maximum are the same between soft and binary centerline for each slices
+    assert (max_coords_over_slices_soft == max_coords_over_slices_bin).all()
```

Comment:
I think that we could probably merge these two tests together?

```suggestion


@pytest.mark.sct_testing
@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
def test_sct_get_centerline_soft_sums_to_one_and_overlaps_with_bin(tmp_path):
    """
    This test checks two necessary conditions of the soft centerline:

    1) The sum of the output intensities of the soft centerline is equal to 1 on all slices
    2) The output maximum of the soft centerline overlaps with the binary segmentation on all slices.
    """
    # Condition 1: All slices of the soft centerline sum to 1
    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '1', '-o',
                                  os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz')])
    im_soft = Image(os.path.join(tmp_path, 't2_seg_centerline_soft.nii.gz'))
    # Sum soft centerline intensities in the (x,y) plane, across all slices
    sum_over_slices = np.apply_over_axes(np.sum, im_soft.data, [0, 2]).flatten()
    # Test if the summation for each slice is equal to 1
    assert (sum_over_slices == 1).all()

    # Condition 2: The max voxels of the soft centerline overlap with the binary centerline
    sct_get_centerline.main(argv=['-i', 't2/t2_seg-manual.nii.gz', '-method', 'fitseg', '-centerline-soft', '0', '-o',
                                  os.path.join(tmp_path, 't2_seg_centerline_bin.nii.gz')])
    im_bin = Image(os.path.join(tmp_path, 't2_seg_centerline_bin.nii.gz'))
    # Find the maximum intensity voxel across all slices in soft centerline and binary centerline
    max_over_slices_soft = np.apply_over_axes(np.max, im_soft.data, [0, 2])
    max_over_slices_bin = np.apply_over_axes(np.max, im_bin.data, [0, 2])
    # Find the coordinates of the maximum in each slice in both soft and binary centerline
    max_coords_over_slices_soft = np.transpose(np.where(im_soft.data == max_over_slices_soft))
    max_coords_over_slices_bin = np.transpose(np.where(im_bin.data == max_over_slices_bin))
    # Test if the maximum are the same between soft and binary centerline for each slices
    assert (max_coords_over_slices_soft == max_coords_over_slices_bin).all()
```

My thinking here is that both tests use the exact same calls to `sct_get_centerline`. So, the actual output file is the same between tests. So, we can check multiple of its properties in one test.